python/lsst/ip/diffim/subtractImages.py

```diff
@@ -853,6 +853,10 @@ def _checkMask(mask, sources, excludeMaskPlanes, checkAdjacent=True):
         xv = (np.rint(sources.getX() - mask.getX0())).astype(int)
         yv = (np.rint(sources.getY() - mask.getY0())).astype(int)
 
+        # The source catalog may not be contiguous.
+        if not sources.isContiguous():
```

Comment:
I'm pretty sure this is too late to fix the problem, since the error report has it coming from the `xv=...` line above. When I said line 854, it must have been on a different version of the file. What I had meant was just after the call to `self.sourceSelector.selectSources` in `_sourceSelector()`. Except `_checkMask` is called on `sources`, not `selected`, so I don't see how it could fail in this way, unless the input catalog is broken somehow?

Reply:
I put the putative fix in `_checkMask` because it operates on the source table in question, but you're probably right that this would best be handled earlier.